frontend/src/concepts/pipelines/content/tables/usePipelineFilter.ts

```diff
@@ -49,9 +80,28 @@ const usePipelineFilter = (
       (key, value) => setFilterData((oldValues) => ({ ...oldValues, [key]: value })),
       [],
     ),
-    onClearFilters: React.useCallback(() => setFilterData(defaultFilterData), []),
+    onClearFilters: React.useCallback(() => {
+      setFilterData(defaultFilterData);
+      if (persistInUrl) {
+        const currentSearchParams = new URLSearchParams(window.location.search);
+        currentSearchParams.delete(FilterOptions.NAME);
+        currentSearchParams.delete(FilterOptions.CREATED_AT);
+        currentSearchParams.delete(FilterOptions.STATUS);
+        currentSearchParams.delete(FilterOptions.EXPERIMENT);
+        currentSearchParams.delete(FilterOptions.PIPELINE_VERSION);
+        navigate(
+          {
+            pathname: window.location.pathname,
+            search: currentSearchParams.toString(),
+          },
+          { replace: true },
+        );
+      }
+    }, [persistInUrl, navigate]),
   };
 
+  const cleanPredicateValue = (value: string) => encodeURIComponent(value.replace(/\\/g, '\\\\'));
```

Comment:
This is not needed any more when you do `encodeURIComponent` on https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/api/pipelines/custom.ts#L45 and change it to `encodeURIComponent(JSON.stringify({ predicates: params.filter.predicates }))`